webapp/tests/data_access/model/test_user.py

```diff
@@ -32,47 +33,44 @@ def test_if_user_generated_then_pdf_is_none(self):
 
 
 class TestUserLastModified(unittest.TestCase):
-    def setUp(self):
-        db.create_all()
+    @pytest.fixture(autouse=True)
+    def attach_fixtures(self, transactional_session):
+        self.session = transactional_session
```

Comment:
To make sure I get it right: This is yet another fixture (but it's not defined in `conftest.py` because we only want to use it here (and it creates an instance variable afterall)). This only uses the `transactional_session` fixture. And pytest only knows that this is the required fixture because  the parameter is also called `transactional_session` like the fixture?
Also, becaus we define it autouse within a class, this is executed before each test within the class (although we do not define the scope as "class"), yes?

Please correct me if anything I said was incorrect.

Reply:
> This is yet another fixture (but it's not defined in conftest.py because we only want to use it here (and it creates an instance variable afterall)). 

Correct! It's only a fixture because that (in combination with autouse) is pytest's way of getting fixtures injected into each test of a unittest class. In "normal" pytest, we would just pass the fixture as a parameter to the test itself. 

> This only uses the transactional_session fixture. 

Correct. 

> And pytest only knows that this is the required fixture because the parameter is also called transactional_session like the fixture?

Correct. This is pytest's dependency injection mechanism, see here: https://docs.pytest.org/en/6.2.x/fixture.html#requesting-fixtures

> Also, becaus we define it autouse within a class, this is executed before each test within the class (although we do not define the scope as "class"), yes?

Yep, correct as well! A `scope=class` fixture is only created once for the class, not for every test in the class: https://docs.pytest.org/en/6.2.x/fixture.html#fixture-scopes

Reply:
Alright, nice, thank you! :)